packages/notifications/src/redux/reducers/notifications.ts

```diff
@@ -0,0 +1,54 @@
+import { PortalNotificationConfig } from '../../Portal';
+import {
+    ADD_NOTIFICATION,
+    REMOVE_NOTIFICATION,
+    CLEAR_NOTIFICATIONS
+} from '../actions/action-types';
+
+function generateID(type: string): string {
+    return 'cloud-services' + type + '-' + new Date().getTime() + Math.random().toString(36).slice(2);
+}
+
+export interface AddNotificationPayload extends Omit<PortalNotificationConfig, 'id'> {
+    id?: string | number;
+}
+
+const addNotification = (notifications: PortalNotificationConfig[], { payload }: { payload: AddNotificationPayload }): PortalNotificationConfig[] =>
+    [ ...notifications, { id: generateID('notification'), ...payload }];
+
+const removeNotification = (notifications: PortalNotificationConfig[], { payload }: { payload: string | number }) => {
```

Comment:
Same as above
```TS
type RemoveNotification = (
  notifications: PortalNotificationConfig[],
  { payload }: { payload: string | number }
) => PortalNotificationConfig[];
```

And use it as
```suggestion
const removeNotification: RemoveNotification = (notifications, { payload }) => {
```